src/main/java/io/vertx/mqtt/messages/MqttPublishMessage.java

```diff
@@ -94,6 +95,11 @@ static MqttPublishMessage create(int messageId, MqttQoS qosLevel, boolean isDup,
    */
   @CacheReturn
   Buffer payload();
+  
+  /**
+   * Send the PUBACK/PUBCOMP to the broker. Use this method only if autoAck option is set to false.
+   */
+  void ack();
```

Comment:
this should throw IllegalStateException if ack was already performed (either once or automatically)

Reply:
good point, I also remove the log and throw an IllegalStateException if you ack a message when Auto Ack is true